src/resources/Pipelines/ResultRankings/ResultRankings.ts

```diff
@@ -51,6 +51,7 @@ export default class ResultRankings extends Resource {
             this.buildPath(ResultRankings.getBaseUrl(pipelineId), {
                 organizationId: this.api.organizationId,
                 ...params,
+                associatedGroups: params?.associatedGroups ? JSON.stringify(params.associatedGroups) : params?.associatedGroups
```

Comment:
In the backend, we expect the array to be serialized as a JSON string. The `buildPath` function internally stringify an array to incompatible formats so I converted the parameter to a JSON string manually.

Reply:
I'm curious, what is the expected format vs the one sent without this change? Do you think there is an option from we could use instead of the workaround you implemented? https://github.com/sindresorhus/query-string#stringifyobject-options

Reply:
The actual behaviour with the current stringify options: `?associatedGroups=group1`

The expected format(JSON): `?associatedGroups=["group1"]`

I've already looked at the different array format options and none of them allow to format as JSON

Reply:
I suspect this is related: https://github.com/sindresorhus/query-string/issues/206 our array is of length one.

Reply:
This is also something I experienced but even tho, I don't see any format that fits my needs 😢

Reply:
Is it a widely supported representation of arrays in query string parameters? I've done some digging and the only ways to represent arrays in URLs are the ones supported by the query-string lib.

Reply:
I would argue that the JSON representation has wider support as it's only a single query parameter that contains a string. Any tool/client will be able to parse the query param and then, they will only need to convert the value to JSON and vice-versa.

As you can see [here](https://swagger.io/docs/specification/serialization/), there's a lot of way to serialize array or objects in a query string and I doubt all http clients support them all out of the box. A JSON string in a URL might not be human friendly but it's easy to use for any developers.

We've used JSON serialization for different different query parameters in our APIs and I don't think we will change that :P

Reply:
In your case you're relying on the parameter being a string, which is a valid format (and I don't think you should break the backend for that) but from my experience `associatedGroups=a&associatedGroups=b` or `associatedGroups=a,b` are more common

I think I understand what you're trying to do, but `JSON.stringify(undefined)` returns `undefined` so the ternary is not useful in this case

```suggestion
                associatedGroups: JSON.stringify(params.associatedGroups)
```

Reply:
Thanks for the explanation @acote-coveo I get it now 👍